backend/models/azfunc.go

```diff
@@ -3,25 +3,57 @@ package models
 import (
 	"encoding/json"
 	"io"
+	"time"
 
 	"github.com/go-playground/validator/v10"
 )
 
 type Query struct {
-	ContinuationToken *string `json:"continuationToken"`
-	MinSize           *uint32 `json:"minSize,string" validation:"omitempty,gt=0"`
-	MaxSize           *uint32 `json:"maxSize,string" validate:"omitempty,gtfieldcustom=MinSize,gt=0"`
-	PageSize          *int    `json:"pageSize,string" validate:"omitempty,gt=0,lte=30"`
+	ContinuationToken    *string      `json:"continuationToken"`
+	PageSize             *int         `json:"pageSize,string" validate:"omitempty,gt=0,lte=30"`
+	Title                *string      `json:"title" validate:"omitempty"`
+	HousingCooperative   *string      `json:"housingCooperative" validate:"omitempty"`
+	ProjectId            *string      `json:"projectId" validate:"omitempty"`
+	PostalCode           *string      `json:"postalCode" validate:"omitempty"`
+	RoomCount            *int         `json:"roomCount,string" validate:"omitempty,gt=0"`
+	MinRoomCount         *int         `json:"minRoomCount,string" validate:"omitempty,gt=0"`
+	MaxRoomCount         *int         `json:"maxRoomCount,string" validate:"omitempty,gt=0,gtfieldcustom=MinRoomCount"`
+	MinSquareMeters      *int         `json:"minSqm,string" validate:"omitempty,gt=0"`
+	MaxSquareMeters      *int         `json:"maxSqm,string" validate:"omitempty,gt=0,gtfieldcustom=MinSquareMeters"`
+	AvailableFrom        *string      `json:"availableFrom" validate:"omitempty,datecustom"`
+	MinYearBuilt         *int         `json:"minYearBuilt,string" validate:"omitempty,gt=1900"`
+	MaxYearBuilt         *int         `json:"maxYearBuilt,string" validate:"omitempty,gt=1900,gtfieldcustom=MinYearBuilt"`
+	MinHwgEnergyClass    *EnergyClass `json:"minHwgEnergyClass" validate:"omitempty,energycustom"`
+	MinFgeeEnergyClass   *EnergyClass `json:"minFgeeEnergyClass" validate:"omitempty,energycustom"`
+	ListingType          *ListingType `json:"listingType" validate:"omitempty,listingtypecustom"`
+	MinRentPricePerMonth *int         `json:"minRentPrice,string" validate:"omitempty,gt=0"`
+	MaxRentPricePerMonth *int         `json:"maxRentPrice,string" validate:"omitempty,gt=0,gtfieldcustom=MinRentPricePerMonth"`
+	MinCooperativeShare  *int         `json:"minCooperativeShare,string" validate:"omitempty,gt=0"`
+	MaxCooperativeShare  *int         `json:"maxCooperativeShare,string" validate:"omitempty,gt=0,gtfieldcustom=MinCooperativeShare"`
+	MinSalePrice         *int         `json:"minSalePrice,string" validate:"omitempty,gt=0"`
+	MaxSalePrice         *int         `json:"maxSalePrice,string" validate:"omitempty,gt=0,gtfieldcustom=MinSalePrice"`
+	SortBy               *string
```

Comment:
SortBy should be validated, otherwise there is a risk of sql injection

Reply:
Now checks if it's exactly one of the fields we allow to sort by. There does not seem to be a way to use the SDK for this since you cannot parametrize ORDER BY and there doesn't seem to be a separate function for validation.